### ISSUE TITLE
Learning rate different in code from paper

### ISSUE THREAD

**Issue by aluo-x** [zekunhao1995/DualSDF#3]
`
The learning rates are 5e-9 for θ,φ and 1e-8 for μ,σ. Each batch consists of 64 shapes; for each shape we sample 2048 SDF values for the high-resrepresentation and 1024 for the primitive-based represen-tation. We train the model for 2800 epochs and drop thelearning rate by 50% after every 700 epochs. We empirically set λ1=λ2= 1e5
`
This is from page 13 of the dualSDF paper on arxiv, but in the code it seems like the initial learning rates for the network is `5e-4`, and the parameters is `0.5`, with a kld weight of `1e-5`. Can authors confirm which is correct?

I believe the actual lr should be the one in the code, except for the parameters (which should be 1e-3, matching DeepSDF). 0.5 just seems like a really high learning rate for Adam. The weighting for the KLD loss I believe should also follow that in the code (1e-5, instead of 1e5).

**Comment by aluo-x** [comment]
Actually I think I understand the reasoning. The numbers given in the paper are scaled relative to the KL loss weight, which if we assume the reconstruction to have weight 1, it would translate to 5e-4 network lr, 1e-3 parameter lr, 1e-5 kld weight. But I still think the current parameter learning rate in the code to mistaken (0.5).

**Comment by zekunhao1995** [comment]
Hi Andrew,

You are right that the learning rates in the paper are scaled to have a D_KL weight of 1. BTW, what is the 0.5 you are referring to i the previous post? Can you point to the code?
Thanks

**Comment by aluo-x** [comment]
I was thinking [this line](https://github.com/zekunhao1995/DualSDF/blob/master/config/dualsdf_chairs_demo.yaml#L71).

**Comment by aluo-x** [comment]
Any idea? I'm assuming the paper is correct and the code is wrong, since after scaling the paper matches the DeepSDF weights.

**Comment by aluo-x** [comment]
@zekunhao1995 can you comment on this? What is the correct value/lr?

**Comment by zekunhao1995** [comment]
Hi Andrew,
Sorry for the late reply. Has been busy with ICCV these days. I have checked the code. The weight for KLD is actually specified [here](https://github.com/zekunhao1995/DualSDF/blob/5882f16cdc41c149cab86215756595de062640fc/config/dualsdf_chairs_demo.yaml#L45). The optimizer learning rate has little to do with the KLD weight as it proportionally scales the gradient from KLD and the gradient from other losses. The reason we use higher learning rate for the latent code is that the gradient to the latent code is much sparser. Hope this helps.
Thanks

**Comment by aluo-x** [comment]
Thanks! I just finished training a VAD-SDF generator based on your code (no primative, just normal clamped deepsdf sample loss) as a baseline, and it works as advertised! 

Best of luck on your ICCV submission!
